primitives.rb

```diff
@@ -0,0 +1,87 @@
+#float/integer block
+@divider = "---------------------------------------------"
+@result_1 = -123.to_i
+@result_2 = '6893'.next.to_i
+@result_3 = 0.to_i-1
+@result_4 = (12.to_f/4.5).ceil.to_i
+@result_5 = ((28%6)*4.7).floor.to_i
+
+def float_integer
+	
+	puts "\n\t ***Homework for lesson4*** \n\n".upcase
+	puts "\ \ I Float/integer"
+	#question 1
+	puts "#1"
+		puts "\ \ -123 it is odd?"
+    	puts "\ \ its #{@result_1.odd?}"
+    	puts @divider
+    #question 2
+    puts "#2"
+    	
+    	puts "\ \ Next number of 6893 will be #{@result_2}"
+    	puts @divider
+    #question 3
+    puts "#3"
+    	
+    	puts "\ \ Previous number of 0 is #{@result_3}"
+    	puts @divider
+    #question 4
+    puts "#4"
+    	
+    	puts "\ \ 12 - cast it to float, then devide by 4.5 then round it up = #{@result_4} "
+    	puts @divider
+    #question 5
+    puts "#5"
+    	
+    	puts "\ \ 28 - call modulo by 6, then multiply result by 4.7 and round it down = #{@result_5}"
+    	puts @divider
+    	
+end
+float_integer
+
+#string block
+puts "\n \ \ II Strings"
+@string_input = "Brand new bicycle"
+
+def strings
+	#question 1
+    puts "#1"
+    	puts "\ \ \"remove trailing special characters\""
+		puts "\ \ #{@string_input}\r\n".delete!("\r\n")
+		puts @divider
+
+	#question 2
+    puts "#2"
+    	puts "\ \ \"Brand new bicycle - using indexes extract new\""
+		puts "\ \ #{@string_input}".match('new')
+		puts @divider
+
+	#question 3
+    puts "#3"
+    	puts "\ \ \"Brand new bicycle - delete last word \""
+    	puts "#{@string_input}"[0...@string_input.rindex(' ')]
+    	puts @divider
+    #question 4
+    puts "#4"
+    	puts "\ \ \"check if the string's last word is bicycle\""
+    	puts "\ \ \"#{@string_input}\" is ending with word \"bicycle\"? - its #{"#{@string_input}".end_with?("bicycle")}"
+    	puts @divider
+    #question 5
+    puts "#5"
+    	puts "\ \ \" Brand new bicycle - find out what index the letter 'w' has\""
+    	puts "index of letter 'w' is #{@string_input.index('w')}" 
+    	puts 
+    	puts @divider
+    #question 6
+    puts "#6"
+    	puts "\ \ \"split the string in 3 parts to get the following result: \"I thought I knew \", \"everything\", \"before that night\" \ \""
+    	puts "I thought I knew everything before that night \n"
+    	puts "I thought I knew everything before that night".split(/(I thought I knew )(everything) (before that night)/)
```

Comment:
This will also return a blank string at the beginning of the array:
"I thought I knew everything before that night".split(/(I thought I knew )(everything) (before that night)/)
=> ["", "I thought I knew ", "everything", "before that night"]